test/e2e/fixtures.go

```diff
@@ -33,6 +33,12 @@ import (
 	"sigs.k8s.io/controller-runtime/pkg/client"
 )
 
+// EchoServerImage is the image to use as a backend fixture.
+// Note that this MUST use a tag, not a digest, in order for
+// the pre-loading into kind to work, since loading + referencing
+// an image by digest is not supported (see https://github.com/kubernetes-sigs/kind/issues/2394).
```

Comment:
Pretty sure this was silently failing previously, i.e. the first E2E to run ended up pulling the image from registry, but with the latest kind it fails loudly when it tries to schedule a pod using the loaded digest. Switching to a tag fixes things.